core/src/main/java/org/lflang/generator/c/CCmakeGenerator.java

```diff
@@ -519,4 +540,134 @@ private static String setUpMainTargetRp2040(
     code.newLine();
     return code.toString();
   }
-}
+
+
+  private static String setUpMainTargetStm32(
+          boolean hasMain, String executableName, Stream<String> cSources, String[] boardProperties) {
+    var code = new CodeBuilder();
+    code.pr("\n# ########################################################");
+    code.pr("\n# ############## [ Start STM32 main target] ##############");
+    code.pr("\n# ########################################################");
+    code.newLine();
+    code.newLine();
+
+    code.pr("add_subdirectory(core)");
+    code.newLine();
+    code.pr("set(LF_MAIN_TARGET " + executableName + ")");
+
+    // Declaration preamble
+    code.pr("set(PROJECT_DIR ${CMAKE_CURRENT_SOURCE_DIR})");
+    code.pr("set(STM_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../STM_sdk)");
```

Comment:
This seems to require that you have a directory called `STM_sdk` in some relative position wrt to src-gen directory? I think we want the user to download the STM SDK themselves and define an environment variable pointing to it

Reply:
Any update here? I think we should require an environmental variable pointing to where STM_sdk is located and the user can then put it anywhere

Reply:
This currently requires a hard-coded location. I was planning on keeping it like this for now, since we also have a sample skeleton that also gives the STM_sdk

Reply:
I dont think this is a very scalable solution. I think we should require an environmental variable called `STM_SDK_PATH` to be set. The generated CMakeLists can verify that it is set and produce an error if not.